ckanext/s3filestore/uploader.py

```diff
@@ -177,7 +180,9 @@ def get_signed_url_to_key(self, key_path):
         if self.download_proxy:
             url = URL_HOST.sub(self.download_proxy + '/', url, 1)
 
-        redis_conn.set(cache_key, '{}|{}'.format(round(current_time), url))
+        if self.is_url_caching_enabled():
+            redis_conn.set(cache_key, url)
```

Comment:
Can we do it in one step as set  has ability to set expiry.

Reply:
redis_conn.set(cache_key, url, ex=self.signed_url_cache_window)

then does not need redis_conn.expire command afterwards.
per docs:

> def set(self, name, value,
> ex=None, px=None, nx=False, xx=False, keepttl=False):
> """
> Set the value at key name to value
> ex sets an expire flag on key name for ex seconds.
> px sets an expire flag on key name for px milliseconds.
> nx if set to True, set the value at key name to value only
> if it does not exist.
> xx if set to True, set the value at key name to value only
> if it already exists.
> keepttl if True, retain the time to live associated with the key.
> (Available since Redis 6.0)